cms_form/models/cms_form_wizard.py

```diff
@@ -0,0 +1,156 @@
+    # display wizard progress bar?
+    wiz_show_progress_bar = True
+    # fields declared here will be automatically stored
+    # into wizard storage
+    # Use `_wiz_step_stored_fields = 'all'` to store them all
+    _wiz_step_stored_fields = []
+
+    @property
+    def _wiz_storage_key(self):
+        """Main storage key."""
+        return self._wiz_name
+
+    @property
+    def _wiz_storage(self):
+        return self.request.session
+
+    def wiz_storage_get(self):
+        if not self._wiz_storage.get(self._wiz_storage_key):
+            # use `deepcopy` to not reference steps' dict
+            self._wiz_storage[self._wiz_storage_key] = \
+                deepcopy(self.DEFAULT_STORAGE_KEYS)
+        return self._wiz_storage[self._wiz_storage_key]
+
+    DEFAULT_STORAGE_KEYS = {
+        'steps': {},
+        'current': 1,
+        'next': None,
+        'prev': None,
+    }
+
+    def form_init(self, request, main_object=None, page=1, wizard=None, **kw):
+        form = super().form_init(request, main_object=main_object, **kw)
+        form.wiz_init(page=page, **kw)
+        return form
+
+    def wiz_init(self, page=1, **kw):
+        steps = self.wiz_configure_steps()
+        storage = self.wiz_storage_get()
+        for k in steps.keys():
+            if k not in storage['steps']:
+                # init missing step
+                storage['steps'][k] = {}
+        current = page
+        storage['current'] = current
+        _next = None
+        if (current + 1) in steps:
+            _next = current + 1
+        _prev = None
+        if (current - 1) in steps:
+            _prev = current - 1
+        storage['next'] = _next
+        storage['prev'] = _prev
+
+    @property
+    def wiz_steps(self):
+        return list(self.wiz_configure_steps().keys())
+
+    def wiz_configure_steps(self):
+        """Configure wizard steps.
+
+        Each step can use a different form, for instance:
+
+        return {
+            1: {
+                'form_model': 'form.a',
+                'title': 'Step 1',
+                'description': 'Preliminary info',
+            },
+            2: {
+                'form_model': 'form.b',
+                'title': 'Step 2',
+            },
+            3: {
+                'form_model': 'form.c',
+                'title': 'Step 3',
+                'description': 'Foo',
+            },
+        }
+        """
+        return {}
+
+    def wiz_get_step_info(self, step):
+        step = int(step)
+        try:
+            return self.wiz_configure_steps()[step]
+        except KeyError:
+            raise ValueError('Step `%d` does not exists.' % step)
+
+    def wiz_current_step(self):
+        return self.wiz_storage_get().get('current') or 1
+
+    def wiz_next_step(self):
+        return self.wiz_storage_get().get('next')
+
+    def wiz_prev_step(self):
+        return self.wiz_storage_get().get('prev')
+
+    def form_next_url(self, main_object=None):
```

Comment:
Here it seems next does not have the same meaning as above at line 112 (wiz_next_step). Not sure if you can use something different like form_following_url ?

Reply:
`form_next_url` comes from cms form base implementation. It stays for "where do you go next after you submit the form".
In wizard case it simply has a similarity w/ `next` action but is not related actually.